ui/components/multichain/network-list-menu/network-list-menu.js

```diff
@@ -85,7 +87,8 @@ export const NetworkListMenu = ({ onClose }) => {
         return null;
       }
 
-      const isCurrentNetwork = currentChainId === network.chainId;
+      const isCurrentNetwork = currentNetwork.id === network.id;
```

Comment:
@darkwing Can you explain how/why this works to solve the issue? I assume that part of the problem is the `selected={isCurrentNetwork}` line below. If two networks within our state have the same `chainId`, wouldn't they have the same `id` property as well?

Reply:
Because every custom and built-in network is given an `id` property.  

Built-in networks: https://github.com/MetaMask/metamask-extension/blob/24c9db588385e9ddb2b41234b27de49f5dcf32fb/ui/selectors/selectors.js#L1241-L1273

Custom networks are all given an `id` as well, which you can see in state's `networkConfigurations` property.  For example, I went to https://chainlist.org/chain/1 and added alternate Mainnets (chain ID 1):

<img width="781" alt="SCR-20230724-nqdc" src="https://github.com/MetaMask/metamask-extension/assets/46655/574d454c-20c4-4df0-906e-98dcf5667152">

Every network `id` is unique, so we don't need to compare chain IDs, which is why my patch works:

https://github.com/MetaMask/metamask-extension/assets/46655/1a042c3c-10d3-4f1f-9c41-acf0f65b800a

Reply:
oh, of coure. I was thinking this was the old `networkId` property, buts its the new unique id per network. Thanks for clarifying that!